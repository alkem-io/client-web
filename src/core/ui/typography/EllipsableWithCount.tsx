import { PropsWithChildren, ReactNode } from 'react';
import { Box, Typography } from '@mui/material';

const EllipsableWithCount = ({ count, children }: PropsWithChildren<{ count: ReactNode }>) => (
  <Box component="span" display="flex" gap={0.5} flexGrow={1} minWidth={0}>
    <Typography component="span" variant="inherit" flexShrink={1} noWrap>
      {children}
    </Typography>
    <Box component="span" flexShrink={0}>
      ({count})
    </Box>
  </Box>
);

export default EllipsableWithCount;
