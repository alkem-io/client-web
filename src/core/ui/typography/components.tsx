import { lighten, Typography, TypographyProps } from '@mui/material';
import provideStaticProps from '@/core/utils/provideStaticProps';

/**
 * @deprecated - the platform header is Alkemio logo
 */
export const PlatformTitle = provideStaticProps(Typography, { variant: 'h1' }) as typeof Typography;

export const PageTitle = provideStaticProps(Typography, { variant: 'h2' }) as typeof Typography;

export const BlockTitle = provideStaticProps(Typography, { variant: 'h3' }) as typeof Typography;

export const BlockSectionTitle = provideStaticProps(Typography, { variant: 'h4' }) as typeof Typography;

export const Tagline = provideStaticProps(Typography, {
  variant: 'subtitle1',
  component: 'h3',
} as TypographyProps) as typeof Typography;

export const Text = provideStaticProps(Typography, { variant: 'body1' }) as typeof Typography;

export const CardTitle = provideStaticProps(Typography, {
  fontWeight: 'bold',
}) as typeof Typography;

export const CardText = provideStaticProps(Typography, {
  variant: 'body2',
  color: theme => lighten(theme.palette.text.primary, 0.4),
}) as typeof Typography;

export const RibbonText = provideStaticProps(Typography, {
  variant: 'body2',
  textAlign: 'center',
  textTransform: 'uppercase',
}) as typeof Typography;

export const Caption = provideStaticProps(Typography, { variant: 'caption', display: 'block' }) as typeof Typography;

export const CaptionSmall = provideStaticProps(Caption, {
  variant: 'caption',
  display: 'block',
  fontStyle: 'italic',
}) as typeof Typography;
