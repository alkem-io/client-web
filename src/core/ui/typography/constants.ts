export const ellipsis = 'â€¦'; // Ellipsis is not just 3 dots
