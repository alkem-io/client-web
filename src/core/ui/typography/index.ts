export * from './components';
export { default as TextBlock } from './TextBlock';
