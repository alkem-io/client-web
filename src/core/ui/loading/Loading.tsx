import { Box, CircularProgress } from '@mui/material';
import { Caption } from '../typography';

export const Loading = ({ text = 'Loading' }: { text?: string }) => {
  return (
    <Box
      sx={{
        display: 'flex',
        flexGrow: 1,
        alignItems: 'center',
        justifyContent: 'center',
        height: '100%',
        gap: 2,
      }}
    >
      <CircularProgress sx={{ color: 'primary.main' }} />
      <Caption textTransform="uppercase" fontWeight="medium" color="primary.main">
        {text}
      </Caption>
    </Box>
  );
};

export default Loading;
