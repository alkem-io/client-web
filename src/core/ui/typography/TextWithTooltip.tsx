import { Tooltip } from '@mui/material';
import { Caption, Text } from '@/core/ui/typography/components';
import React from 'react';

const TextWithTooltip = ({ text, tooltip }: { text: string; tooltip: string }) => {
  return (
    <Tooltip arrow title={<Caption>{tooltip}</Caption>} placement="top">
      <Text>{text}</Text>
    </Tooltip>
  );
};

export default TextWithTooltip;
