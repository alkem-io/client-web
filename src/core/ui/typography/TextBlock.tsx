import { Box, BoxProps } from '@mui/material';
import { rem } from './utils';
import { GUTTER_PX } from '../grid/constants';

const TextBlock = (props: BoxProps) => {
  return <Box {...props} display="flex" flexDirection="column" gap={rem(GUTTER_PX / 2)} />;
};

export default TextBlock;
