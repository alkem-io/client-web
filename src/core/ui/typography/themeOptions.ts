import { TypographyOptions } from '@mui/material/styles/createTypography';
import { gutter } from '../grid/constants';
import { rem } from './utils';

export const fontFamilyMontserrat = '"Montserrat", sans-serif';
export const fontFamilySourceSans = '"Source Sans Pro", sans-serif';

const fontWeightRegular = 400;
const fontWeightMedium = 500;
const fontWeightBold = 700;

const lineHeightSingle = rem(gutter);
const lineHeightDouble = rem(gutter * 2);

export const typographyOptions: TypographyOptions = {
  h1: {
    fontFamily: fontFamilyMontserrat,
    fontSize: rem(25),
    fontWeight: fontWeightBold,
    lineHeight: lineHeightDouble,
  },
  h2: {
    fontFamily: fontFamilyMontserrat,
    fontSize: rem(18),
    fontWeight: fontWeightBold,
    lineHeight: lineHeightSingle,
  },
  h3: {
    fontFamily: fontFamilyMontserrat,
    fontSize: rem(15),
    fontWeight: fontWeightRegular,
    lineHeight: lineHeightSingle,
  },
  h4: {
    fontFamily: fontFamilyMontserrat,
    fontSize: rem(12),
    fontWeight: fontWeightRegular,
    lineHeight: lineHeightSingle,
  },
  subtitle1: {
    fontFamily: fontFamilyMontserrat,
    fontSize: rem(16),
    fontWeight: fontWeightRegular,
    lineHeight: lineHeightSingle,
    fontStyle: 'italic',
  },
  body1: {
    fontFamily: fontFamilySourceSans,
    fontSize: rem(14),
    fontWeight: fontWeightRegular,
    lineHeight: lineHeightSingle,
  },
  body2: {
    fontFamily: fontFamilySourceSans,
    fontSize: rem(12),
    fontWeight: fontWeightRegular,
    lineHeight: lineHeightSingle,
  },
  button: {
    fontFamily: fontFamilyMontserrat,
    fontSize: rem(12),
    fontWeight: fontWeightMedium,
    lineHeight: lineHeightSingle,
    textTransform: 'uppercase',
  },
  caption: {
    fontFamily: fontFamilyMontserrat,
    fontSize: rem(12),
    fontWeight: fontWeightRegular,
    lineHeight: lineHeightSingle,
  },
};
