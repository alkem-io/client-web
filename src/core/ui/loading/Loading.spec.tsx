/**
 * @jest-environment jsdom
 */
import { render, screen } from '@/main/test/testUtils';
import { describe, expect, test } from 'vitest';
import { Caption } from '../typography';
import Loading from './Loading';

describe('Loading component', () => {
  // Mathcing snapshot when we are using dynamicaly generated class names is ineffective
  test.skip('render correctly WrapperTypography component', () => {
    // arrange
    const { asFragment } = render(<Caption textTransform="uppercase" fontWeight="medium" color="primary.main" />);

    // act
    const html = asFragment();

    // assert
    expect(html).toMatchSnapshot(); // Mathcing snapshot when we are using dynamicaly generated class names is ineffective
  });

  // Mathcing snapshot when we are using dynamicaly generated class names is ineffective
  test.skip('render correctly loading component', () => {
    // arrange
    const text = 'Please wait';

    // act
    const { asFragment } = render(<Loading text={text} />);

    // assert
    expect(asFragment()).toMatchSnapshot(); // Mathcing snapshot when we are using dynamicaly generated class names is ineffective
  });

  test.skip('check loading with message', async () => {
    // arrange
    const message = 'Loading indicator';

    // act
    render(<Loading text={message} />);

    // assert
    expect(screen.getByText(message)).toBeInTheDocument();
  });
});
