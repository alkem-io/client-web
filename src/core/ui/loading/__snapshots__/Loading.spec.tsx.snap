// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component render correctly WrapperTypography component 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-caption-6 makeStyles-primary-9"
    style="font-weight: 500;"
  />
</DocumentFragment>
`;

exports[`Loading component render correctly loading component 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-container-20"
  >
    <div
      class="MuiCircularProgress-root makeStyles-spinner-18 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
    <div
      class="makeStyles-caption-26 makeStyles-primary-29 makeStyles-text-19"
      style="font-weight: 500;"
    >
      <span>
        Please wait
      </span>
    </div>
  </div>
</DocumentFragment>
`;
